app/src/main/java/com/example/quiz/MainActivity.java

```diff
@@ -42,55 +42,52 @@ public class MainActivity extends AppCompatActivity {
     private int userScore;
     private TextView totalQuestions;
     final int USER_PROGRESS = (int) Math.ceil(100.0 / (double) 10); //increments progress bar by USER_PROGRESS each time answered
-    private int mQuestionIndex;
+    private int mQuestionIndex = 0;
     private String mQuizQuestion;
     private List<QuizQuestion> questionCollection;
 
+    private View loadingView;
```

Comment:
Loading view added with new layout to indicate that the questions are being loaded for a better user experience.